app/core/events.py

```diff
@@ -9,6 +9,23 @@
 
 def create_start_app_handler(app: FastAPI):
     def start_app() -> None:
+        container_name, new_container_name = utils.generate_container_name("premd")
+        client = utils.get_docker_client()
+        if utils.container_exists(container_name):
+            container = client.containers.get(container_name)
+            host_port = container.ports.get(f"{utils.DEFAULT_PORT}/tcp", [None])[0][
+                "HostPort"
+            ]
+            if host_port != f"{utils.DEFAULT_PORT}":
+                utils.check_host_port_availability(utils.DEFAULT_PORT)
```

Comment:
What if this returns `False`? If you ignore the return value of this function, why call it at all?